ui/redux/selectors/search.js

```diff
@@ -60,34 +61,54 @@ export const makeSelectHasReachedMaxResultsLength = (query: string): ((state: St
 
 export const makeSelectRecommendedContentForUri = (uri: string) =>
   createSelector(
-    makeSelectClaimForUri(uri),
+    selectHistory,
+    selectClaimsByUri,
     selectSearchResultByQuery,
     makeSelectClaimIsNsfw(uri),
-    (claim, searchUrisByQuery, isMature) => {
+    (history, claimsByUri, searchUrisByQuery, isMature) => {
+      const claim = claimsByUri[uri];
+
+      if (!claim) return;
+
       let recommendedContent;
-      if (claim) {
-        // always grab full URL - this can change once search returns canonical
-        const currentUri = buildURI({ streamClaimId: claim.claim_id, streamName: claim.name });
+      // always grab the claimId - this value won't change for filtering
+      const currentClaimId = claim.claim_id;
 
-        const { title } = claim.value;
+      const { title } = claim.value;
 
-        if (!title) {
-          return;
-        }
+      if (!title) return;
 
-        const options: {
-          related_to?: string,
-          nsfw?: boolean,
-          isBackgroundSearch?: boolean,
-        } = { related_to: claim.claim_id, isBackgroundSearch: true };
+      const options: {
+        related_to?: string,
+        nsfw?: boolean,
+        isBackgroundSearch?: boolean,
+      } = { related_to: claim.claim_id, nsfw: isMature, isBackgroundSearch: true };
```

Comment:
I know this result is the same as previous, but I think we don't necessarily want isMature to be set TRUE in search just because the claim is. @tzarebczan

Reply:
that's correct, respect the mature content setting.

Reply:
Yeah that makes sense, specially since you can't even see the content it wouldn't make sense to display nsfw on the sidebar... So I did some changes to the search based on the user setting instead of doing a regular search and filtering later, which would cause only a few videos or none to show up instead of all 20, the results are pretty interesting lol:

![Screenshot_2021-09-16_09-50-00](https://user-images.githubusercontent.com/76502841/133615274-fe809180-8f7f-4fd1-b948-41da4c9874b7.png)